pkg/containerutil/cp_linux.go

```diff
@@ -237,97 +277,74 @@ func CopyFiles(ctx context.Context, client *containerd.Client, container contain
 		return err
 	}
 	tarXCmd.Stdout = os.Stderr
-	tarXCmd.Stderr = os.Stderr
+	var tarErr bytes.Buffer
+	tarXCmd.Stderr = &tarErr
 
 	log.G(ctx).Debugf("executing %v in %q", tarCCmd.Args, tarCCmd.Dir)
 	if err := tarCCmd.Start(); err != nil {
-		return fmt.Errorf("failed to execute %v: %w", tarCCmd.Args, err)
+		return errors.Join(fmt.Errorf("failed to execute %v", tarCCmd.Args), err)
 	}
+
 	log.G(ctx).Debugf("executing %v in %q", tarXCmd.Args, tarXCmd.Dir)
 	if err := tarXCmd.Start(); err != nil {
-		return fmt.Errorf("failed to execute %v: %w", tarXCmd.Args, err)
+		if strings.Contains(err.Error(), "permission denied") {
+			return ErrTargetIsReadOnly
+		}
+
+		// Other errors, just put them back on stderr
+		_, fpErr := fmt.Fprint(os.Stderr, tarErr.String())
+		if fpErr != nil {
+			return errors.Join(fpErr, err)
+		}
+
+		return errors.Join(fmt.Errorf("failed to execute %v", tarXCmd.Args), err)
```

Comment:
we can just use `fmt.Errorf("failed to execute %v: %w", tarXCmd.Args, err)`

Reply:
I think `errors.Join` is better.

With `fmt.Errorf`:
- we get a variety of format accross the codebase ("failed: %w", "failed (%w)", "failed with %w", "%w (hint: foo bla)")
- because of ^, things are less readable, as errors are clumped together with little indication which comes first